django_lightweight_queue/middleware/transaction.py

```diff
@@ -1,14 +1,29 @@
 from django.db import transaction, connection
 
+from ..task import task
+
 class TransactionMiddleware(object):
+    _transaction_started = False
+
+    def _enable_transaction(job):
+        return job.get_middleware_option('transaction.enabled', True)
+
     def process_job(self, job):
-        transaction.atomic().__enter__()
+        if _enable_transaction(job):
+            transaction.atomic().__enter__()
+            self._transaction_started = True
 
     def process_result(self, job, result, duration):
-        transaction.atomic().__exit__(None, None, None)
+        if self._transaction_started:
+            transaction.atomic().__exit__(None, None, None)
 
     def process_exception(self, job, time_taken, *exc_info):
-        transaction.atomic().__exit__(*exc_info)
+        if self._transaction_started:
+            transaction.atomic().__exit__(*exc_info)
+
+def non_transactional_task(*args, **kwargs):
+    kwargs.setdefault('transaction', {}).setdefault('enabled', False)
+    return task(*args, **kwargs)
```

Comment:
I'm not sure I like the pattern of making this a specific version of `task`–if another middleware had a similar mechanism (`unlogged_task` for the sake of argument) one wouldn't be able to combine the decorators.